src/compiler/checker.ts

```diff
@@ -3302,7 +3302,7 @@ namespace ts {
                 function writeTypeReference(type: TypeReference, flags: TypeFormatFlags) {
                     const typeArguments = type.typeArguments || emptyArray;
                     if (type.target === globalArrayType && !(flags & TypeFormatFlags.WriteArrayAsGenericType)) {
-                        writeType(typeArguments[0], TypeFormatFlags.InElementType);
+                        writeType(typeArguments[0], TypeFormatFlags.InElementType | TypeFormatFlags.InArrayType);

```

Comment:
how does `InElementType` differ from `InArrayType`? The descriptions make them seem like `InArrayType1`, `InArrayType2` would be better names.